components/hab/src/main.rs

```diff
@@ -523,7 +523,13 @@ fn sub_bldr_job_status(ui: &mut UI, m: &ArgMatches) -> Result<()> {
     let url = bldr_url_from_matches(m);
     let group_id = m.value_of("GROUP_ID");
     let origin = m.value_of("ORIGIN");
-    command::bldr::job::status::start(ui, &url, group_id, origin)
+    let limit = m.value_of("LIMIT")
+        .unwrap_or("10")
+        .parse::<usize>()
+        .unwrap();
```

Comment:
Per @elliott-davis's comment above, it seem like it would be nice to omit the checking in the CLI macro in favor of doing it all here. We could make this
```rust
let limit = m.value_of("LIMIT").unwrap_or("10").parse::<usize>()?;
```
if we add
```rust
impl From<num::ParseIntError> for Error {
    fn from(err: num::ParseIntError) -> Self {
        Error::ParseIntError(err)
    }
}
```
to [errors.rs](https://github.com/habitat-sh/habitat/blob/SA/update-job-status/components/hab/src/error.rs). It already has a `ParseIntError` in the enum, so I'm not sure why we don't have this.

Reply:
At some point, we decided to move away from using the From pattern for errors, and use map_err instead for more transparency.  Either would be fine, it's nice to have the check up front though so I'l just leave it this way for now.